src/Core/Services/Fido2ClientService.cs

```diff
@@ -109,16 +109,20 @@ public async Task<Fido2ClientCreateCredentialResult> CreateCredentialAsync(Fido2
                 throw new Fido2ClientException(Fido2ClientException.ErrorCode.NotSupportedError, "No supported algorithms found");
             }
 
-            var clientDataJSON = JsonSerializer.Serialize(new
+            byte[] clientDataJSONBytes = null;
+            if (clientDataHash == null)
             {
```

Comment:
🤔 Is it correct that if `clientDataHash` is **not** `null` then `clientDataJSONBytes` is `null`?
Same In `AssertCredentialAsync(...)`

Reply:
Yes because they are derived from each other `ClientDataJSON -> ClientDataJSONBytes -> ClientDataHash`. So if `Fido2Client` gets a `ClientDataHash` as a parameter then it won't know what that has was calculated from (Android/Browser does this for us)